2023-02-03 build failure with USE_SYSTEM_LIBLUA

It seems like the include path LuaShip.cpp hardcodes path to bundled lua, while the packages are always built with system lua.

```
/work/usr/ports/games/pioneer/work/pioneer-20230203/src/lua/LuaShip.cpp:22:10: fatal error: 'src/lua.h' file not found
#include "src/lua.h"
         ^~~~~~~~~~~
1 warning and 1 error generated.
```

--- a/src/lua/LuaShip.cpp
+++ b/src/lua/LuaShip.cpp
@@ -19,7 +19,7 @@
 #include "SpaceStation.h"
 #include "ship/PlayerShipController.h"
 #include "ship/PrecalcPath.h"
-#include "src/lua.h"
+#include "lua.h"
 
 /*
  * Class: Ship

